nipoppy/workflows/pipeline.py

```diff
@@ -163,12 +164,86 @@ def fpath_container(self) -> Path:
             )
 
         elif not fpath_container.exists():
-            raise FileNotFoundError(
-                f"No container image file found at {fpath_container} for pipeline"
-                f" {self.pipeline_name} {self.pipeline_version}"
-            )
+            try:
+                self.logger.warning(
+                    f"No container image file found at {fpath_container} for pipeline"
+                    f" {self.pipeline_name} {self.pipeline_version}.\n"
+                )
+                self._get_container_image()
+            except:  # noqa: E722
+                raise FileNotFoundError(
+                    f"No container image file found at {fpath_container} for pipeline"
+                    f" {self.pipeline_name} {self.pipeline_version}"
+                )
         return fpath_container
 
+    def _get_container_image(self):
+        self.logger.info(
+            "Getting container image for "
+            f"{self.pipeline_name} {self.pipeline_version} "
+            f"at {self.fpath_container}"
+        )
+
+        if self.pipeline_name in ["mriqc", "fmriprep"]:
+            docker_hub_repo = "nipreps"
+        elif self.pipeline_name in ["dcm2bids"]:
+            docker_hub_repo = "nipreps"
+        elif self.pipeline_name in ["bidscoin"]:
+            docker_hub_repo = "nipreps"
+        elif self.pipeline_name in ["heudiconv"]:
+            docker_hub_repo = "nipreps"
+
+        command = (
+            f"{self.CONTAINER_CONFIG.COMMAND} build {self.fpath_container} "
+            f"docker://{docker_hub_repo}/{self.pipeline_name}:{self.pipeline_version}"
+        )
+
+        self.log_command(command)
+
+        def process_output(
```

Comment:
copy pasted from the base workflow

may need to extract it into a separate method to be able to run it

Reply:
Should this not just be a call to `self.run_command(command)`? Which will log the command and run it in a subprocess

https://github.com/nipoppy/nipoppy/blob/379ace12e40369af0655670a63553cad942a2d50/nipoppy/workflows/base.py#L95-L182